staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -564,15 +564,21 @@ func (f *featureGate) featureSpec(key Feature) (FeatureSpec, error) {
 }
 
 func (f *featureGate) unsafeRecordQueried(key Feature) {
-	queriedFeatures := map[Feature]struct{}{}
-	for k := range f.queriedFeatures.Load().(map[Feature]struct{}) {
-		queriedFeatures[k] = struct{}{}
-	}
-	if _, ok := queriedFeatures[key]; ok {
+	queriedFeatures := f.loadQueriedFeatures()
+	if queriedFeatures.Has(key) {
 		return
 	}
-	queriedFeatures[key] = struct{}{}
-	f.queriedFeatures.Store(queriedFeatures)
+	// Clone queriedFeatures only when it needs to be mutated
```

Comment:
Nit: If I am understanding it, this is the key optimization, not the switch from `atomic.Value` to `atomic.Pointer` for the map. Could you update the commit message and PR text please? Is there a good reason to keep that part of the change?

Reply:
Right, `atomic.Pointer` and `set` makes it a bit more explicit (and easier to understand the purpose of this function imo), but to keep the change minimal I reworked it again so that the change would focus on "copy only when mutation is needed".
@siyuanfoundation would you like to have another look too in case I missed something else? Thank you

Reply:
I think it's ok to change the map to set since it is more generic and explicit.